File: sdks/official/fraiseql-php/src/Exceptions/GraphQLException.php
<?php

declare(strict_types=1);

namespace FraiseQL\Exceptions;

class GraphQLException extends FraiseQLException
{
    /** @param list<array{message: string}> $errors */
    public function __construct(
        public readonly array $errors,
        string $message = ''
    ) {
        parent::__construct($message ?: ($errors[0]['message'] ?? 'GraphQL error'));
    }
}

File: sdks/official/fraiseql-java/src/main/java/io/fraiseql/GraphQLException.java
package io.fraiseql;

import java.util.Collections;
import java.util.List;

/** Thrown when the GraphQL response contains one or more errors. */
public class GraphQLException extends FraiseQLException {
    private final List<GraphQLError> errors;

    public GraphQLException(List<GraphQLError> errors) {
        super(errors.isEmpty() ? "GraphQL error" : errors.get(0).getMessage());
        this.errors = Collections.unmodifiableList(errors);
    }

    public List<GraphQLError> getErrors() { return errors; }
}
